diff -up qtxmlpatterns-everywhere-src-5.15.0-alpha/src/xmlpatterns/parser/qquerytransformparser.cpp.1~ qtxmlpatterns-everywhere-src-5.15.0-alpha/src/xmlpatterns/parser/qquerytransformparser.cpp
--- qtxmlpatterns-everywhere-src-5.15.0-alpha/src/xmlpatterns/parser/qquerytransformparser.cpp.1~	2020-02-17 22:42:04.163002764 +0100
+++ qtxmlpatterns-everywhere-src-5.15.0-alpha/src/xmlpatterns/parser/qquerytransformparser.cpp	2020-02-17 22:42:45.772004599 +0100
@@ -1499,7 +1499,7 @@ typedef short yytype_int16;
 # define YYUSE(E) /* empty */
 #endif
 
-#if defined __GNUC__ && ! defined __ICC && 407 <= __GNUC__ * 100 + __GNUC_MINOR__
+#if !defined(__clang__) && defined __GNUC__ && ! defined __ICC && 407 <= __GNUC__ * 100 + __GNUC_MINOR__
 /* Suppress an incorrect diagnostic about yylval being uninitialized.  */
 # define YY_IGNORE_MAYBE_UNINITIALIZED_BEGIN \
     _Pragma ("GCC diagnostic push") \
